opm/io/eclipse/EclFile.hpp

```diff
@@ -118,4 +124,68 @@ class EclFile
 
 }} // namespace Opm::EclIO
 
+
+namespace Opm {
+    namespace EclIO {
+    
+    class EclFileLGR : public EclFile {
+    public:
+        EclFileLGR(const std::string& filename, bool preload = false)
+            : EclFile(filename, preload)
+        {
+            initialize_lgr_properties();
+            classify_lgr();
+        };
+    
+        EclFileLGR(const std::string& filename, Formatted fmt, bool preload = false)
+            : EclFile(filename, fmt, preload)
+        {
+            initialize_lgr_properties();
+            classify_lgr();
+        };
+        
+        template <typename T>
+        const std::vector<T>& get(int arrIndex)
+        {
+            return EclFile::get<T>(arrIndex);
+        };
+      
+        template <typename T>
+        const std::vector<T>& get(const std::string& name){
+            return EclFile::get<T>(name);
+        };
+        template <typename T>
+        const std::vector<T>& get(const std::string& name, const std::string& lgr_name = "GLOBAL")
+        {
+            const auto tag = array_name_to_num[name];
+            const int lgr_num = lgr_labels[lgr_name];
+            for (std::size_t idx = 0; idx < lgr_array_index.size(); ++idx) {
+                if (lgr_array_index[idx] == tag && lgr_classification[idx] == lgr_num) {
+                    return get<T>(idx);
+                }
+            }
+            throw std::invalid_argument("No array found for name '" + name + "' and lgr '" + lgr_name + "'");
+
+        };
+
+
+    protected:
```

Comment:
Do you intend for other classes to inherit from `EclFileLGR`?  If not, these data members should typically be marked `private:` instead.